integrations/weaviate/tests/test_bm25_retriever.py

```diff
@@ -10,6 +10,7 @@ def test_init_default():
     assert retriever._document_store == mock_document_store
     assert retriever._filters == {}
     assert retriever._top_k == 10
+    assert retriever._filter_policy == "replace"
```

Comment:
I don't see any tests for the `merge` policy?

Reply:
Yes @shadeMe I just coordinated with @anakin87 about the same topic in https://github.com/deepset-ai/haystack-core-integrations/pull/820 as now we need to add essentially the same test for all integrations - a bit unnecessary. 

Perhaps I can introduce the use of now available FilterPolicy Enum in all  relevant integrations and we can have one merge unit test (as @anakin87 suggested) for essentially this:

```
        if self.filter_policy == "merge" and filters:
            filters = {**self.filters, **filters}
        else:
            filters = filters or self.filters
```

in Haystack at one place rather than needlessly add bunch of identical copy/paste tests all over integrations. 
WDYT?

Reply:
Don't we have a test suite for `DocumentStore`s that we share across all integrations? Let's add it there?